### PR TITLE
Allow to install symfony/* 5.x dependencies

### PR DESCRIPTION
**Pull request by phansys**
## Subject
Allow to install 5.x versions for `symfony/browser-kit`, `symfony/config`, `symfony/css-selector`, `symfony/dependency-injection`, `symfony/http-foundation`, `symfony/intl`, `symfony/options-resolver`, `symfony/phpunit-bridge` and `symfony/templating`.

I am targeting this branch, because these changes respect BC.

## Changelog

```markdown
### Added
- Support for 5.x versions in `symfony/browser-kit`, `symfony/config`, `symfony/css-selector`, `symfony/dependency-injection`, `symfony/http-foundation`, `symfony/intl`, `symfony/options-resolver`, `symfony/phpunit-bridge` and `symfony/templating`.
```

### CODE REVIEW

**Comment by phansys** [comment]
Even if there are dependencies that aren't resolved to the 5.x version (`symfony/form`, `symfony/framework-bundle`), I think we could keep an eye on them when the other conflicting dependencies blocking these updates are resolved.

**Comment by greg0ire** [comment]
@phansys isn't it because Sonata[DoctrineORM]AdminBundle 4 are not allowed yet? Shouldn't we start with these packages?

**Comment by phansys** [comment]
Yes, `sonata-project/admin-bundle` and `sonata-project/doctrine-orm-admin-bundle`, by instance:
```
composer why-not symfony/form ^5
sonata-project/admin-bundle               3.106.1  requires  symfony/form (^4.4.12)  
sonata-project/doctrine-orm-admin-bundle  3.35.0   requires  symfony/form (^4.4)
```
I removed the `^5.3` range for these dependencies, so we are safe about all the resolvable versions as they are covered by our CI.

**Review by greg0ire — DISMISSED** [review]

File: composer.json [hunk 1]
```diff
@@ -23,13 +23,13 @@
...
-        "symfony/dependency-injection": "^4.4",
+        "symfony/config": "^4.4 || ^5.3",
+        "symfony/dependency-injection": "^4.4 || ^5.3",
         "symfony/form": "^4.4",
-        "symfony/http-foundation": "^4.4",
-        "symfony/http-kernel": "^4.4",
-        "symfony/intl": "^4.4",
-        "symfony/options-resolver": "^4.4",
+        "symfony/http-foundation": "^4.4 || ^5.3",
+        "symfony/http-kernel": "^4.4 || ^5.3",
```
Comment (jordisala1991) [inline composer.json]: this is not getting installed for 5.3

Comment (phansys) [reply composer.json]: You're right. Fixed:
```
composer show symfony/* --direct
symfony/browser-kit          v5.3.4  Simulates the behavior of a web browser, allowing you to make requests, click on links and submit forms programmatically
symfony/config               v5.3.10 Helps you find, load, combine, autofill and validate configuration values of any kind
symfony/css-selector         v5.3.4  Converts CSS selectors to XPath expressions
symfony/dependency-injection v5.3.10 Allows you to standardize and centralize the way objects are constructed in your application
symfony/form                 v4.4.33 Allows to easily create, process and reuse HTML forms
symfony/framework-bundle     v4.4.31 Provides a tight integration between Symfony components and the Symfony full-stack framework
symfony/http-foundation      v5.3.10 Defines an object-oriented layer for the HTTP specification
symfony/http-kernel          v4.4.33 Provides a structured process for converting a Request into a Response
symfony/intl                 v5.3.8  Provides a PHP replacement layer for the C intl extension that includes additional data from the ICU library
symfony/options-resolver     v5.3.7  Provides an improved replacement for the array_replace PHP function
symfony/phpunit-bridge       v5.3.10 Provides utilities for PHPUnit, especially user deprecation notices management
symfony/templating           v5.3.7  Provides all the tools needed to build any kind of template system
```

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
It would be nice to make the changelog more explicit, since not all symfony components are compatible with sf 5.3 (not a blocking point to merge)

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**